Files/Files.csproj

```diff
@@ -328,7 +328,9 @@
     <Compile Include="UserControls\Widgets\WidgetsListControl.xaml.cs">
       <DependentUpon>WidgetsListControl.xaml</DependentUpon>
     </Compile>
+    <Compile Include="ViewModels\NavToolbarViewModel.cs" />
```

Comment:
What value does the view model offer over having the events in the code behind of the control?

Reply:
This way each shell page can only effect it's own nav toolbar view model, so if multiple pages are loading at the same time, they won't both be making conflicting changes to the toolbar ui at once. It also tabs/panes from responding to events when they aren't selected without unsubscribing and subscribing to the events every time the selected page is changed.

Reply:
What is the performance implications of doing this? Are the ViewModels completely destroyed when a tab is closed?

Reply:
Atm, they should be destroyed when ``ModernShellPage`` gets gced, however since there's a bit of an issue with that I'll destroy them manually in the ``Dispose`` function.